### PR TITLE
Feature/currency

### PR DESCRIPTION
**Pull request by tomzweb**

### CODE REVIEW

File: frontend/resources/js/helpers/configuration.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import axios from 'axios';
+import { currencySelector } from './elements';
+
+const getCurrencySelectorCurrencyCode = () => {
+  // the only way I can see to get the active currency selector is by finding the bold item in the dropdown,
+  // then using the data attribute to get the code
+  const option = currencySelector?.querySelector('strong')?.parentElement;
```
Comment (bobbyshaw) [inline frontend/resources/js/helpers/configuration.ts:7]: From what I can tell this doesn't work on ragstock so is probably a bit too brittle. They do have the strong element but they've removed the currency code data attribute. 

The only other place that I can see it in the HTML is here on cornerstone:

```
<main class="body" id="main-content" role="main" data-currency-code="USD">
```

The ragstock theme has the same element albeit pretty different markup, it still has the body class:

```
<div class="body content" data-currency-code="EUR">
```

document.querySelector('.body')?.dataset?.currencyCode

It does seem to be stored in this cookie. It wouldn't be pretty getting it out here either. The only plus is that it's controlled by BC it's perhaps less likely to break the DOM selector.  (Cur: CAD)

Screenshot 2023-01-16 at 15 31 46

File: frontend/resources/js/helpers/price.ts [hunk 2]
```diff
@@ -1 +1,25 @@
...
+  const thouSep = currencyInfo.display.thousandsToken ?? ',';
+  const decSep = currencyInfo.display.decimalToken ?? '.';
+
+  const sign = number < 0 ? '-' : '';
+  // eslint-disable-next-line radix
+  const i = String(parseInt(number = Math.abs(Number(number) || 0).toFixed(decPlaces)));
+  // @ts-ignore
+  var j = (j = i.length) > 3 ? j % 3 : 0;
+
+  return sign +
```
Comment (bobbyshaw) [inline frontend/resources/js/helpers/price.ts:20]: Do I spy a stackoverflow answer? :D

File: frontend/resources/js/helpers/configuration.ts (hunk 1 above)
Comment (tomzweb) [reply frontend/resources/js/helpers/configuration.ts:7]: Nice, let me investigate.

File: frontend/resources/js/helpers/price.ts (hunk 2 above)
Comment (tomzweb) [reply frontend/resources/js/helpers/price.ts:20]: Oh yes 🤣

**Pull request merged**